modules/internal-load-balancer/main.tf

```diff
@@ -1,96 +1,61 @@
-  }
-  timeout_sec = 4
+  address_type = "INTERNAL"
+  description  = "The private IP address for the TFE internal load balancer."
+  subnetwork   = var.vpc_subnetwork_self_link
 }
 
-resource "google_compute_region_backend_service" "application" {
-  health_checks = [google_compute_region_health_check.application.self_link]
-  name          = local.application_name
-
-  dynamic "backend" {
-    for_each = var.primaries_instance_groups_self_links
-    content {
-      group = backend.value
-
-      balancing_mode        = "RATE"
-      capacity_scaler       = 1.0
-      description           = "Some of the TFE primaries."
-      max_rate_per_instance = 333
+resource "google_compute_instance" "main" {
+  boot_disk {
+    initialize_params {
+      image = var.disk_image
+      size  = var.disk_size
+      type  = "pd-ssd"
     }
   }
-  backend {
-    group = var.secondaries_instance_group_manager_instance_group
+  machine_type = var.machine_type
+  name         = local.name
+  network_interface {
+    subnetwork         = var.vpc_subnetwork_self_link
+    subnetwork_project = var.vpc_subnetwork_project
 
-    balancing_mode        = "RATE"
-    capacity_scaler       = 1.0
-    description           = "The TFE secondaries."
-    max_rate_per_instance = 333
+    network_ip = google_compute_address.main.address
   }
-  description           = "TFE application."
-  load_balancing_scheme = "INTERNAL_MANAGED"
-  port_name             = "application"
-  protocol              = "HTTPS"
-  timeout_sec           = 10
-}
-
-resource "google_compute_region_url_map" "application" {
-  name = local.application_name
-
-  default_service = google_compute_region_backend_service.application.self_link
-  description     = "TFE application requests."
-}
 
-resource "google_compute_region_ssl_certificate" "application" {
-  certificate = var.ssl_certificate
-  private_key = var.ssl_certificate_private_key
-
-  description = "TFE application."
-  name_prefix = var.prefix
-}
-
-resource "google_compute_region_target_https_proxy" "application" {
-  name             = local.application_name
-  ssl_certificates = [google_compute_region_ssl_certificate.application.self_link]
-  url_map          = google_compute_region_url_map.application.self_link
-
-  description = "TFE application traffic."
-}
-
-resource "google_compute_address" "main" {
-  name = local.application_name
+  allow_stopping_for_update = true
+  description               = "TFE internal load balancer."
+  labels                    = var.labels
+  metadata_startup_script = templatefile(
+    "${path.module}/templates/startup.sh.tmpl",
+    {
+      address                        = google_compute_address.main.address
+      primary_0_address              = var.primaries_instances_network_interface_network_ips[0]
+      primary_1_address              = var.primaries_instances_network_interface_network_ips[1]
+      primary_2_address              = var.primaries_instances_network_interface_network_ips[2]
```

Comment:
how is this impacted if someone needs to roll out a new primary to replace one of the existing ones?

Reply:
The `metadata_startup_script` attribute will cause the compute instance to be recreated if it changes. Alternatively, we could reserve the IP addresses of the compute instances so this wouldn't have to change at all, which seems more desirable now that I've written it down. :grin:

Reply:
@erindatkinson https://github.com/hashicorp/terraform-google-terraform-enterprise/pull/74/commits/58bd2ea5c5973cc226a3f760808608dd30ca6aa5 changed the design so that the primary addresses are static. I tested the load balancer using the example while primary-2 was being recreated and it still served application traffic.